packages/uicore/src/components/MultiSelectDropDown/MultiSelectDropDown.tsx

```diff
@@ -102,12 +104,22 @@ export function MultiSelectDropDown(props: MultiSelectDropDownProps): React.Reac
     }
   }, [JSON.stringify(items)])
 
-  React.useEffect(() => {
+  useEffect(() => {
     if (Array.isArray(value)) {
       setSelectedItems(value)
     }
   }, [value])
 
+  const onSearchChange = useCallback(
+    (newQuery: string) => {
+      if (expandingSearchInputProps && 'onChange' in expandingSearchInputProps) {
```

Comment:
had to do an additional check here otherwise TS will give a warning about unable to use `onChange in` on expandingSearchInputProps being potentially undefined

Reply:
Ah I came across this absolute nightmare too, you can change the `onChange` to default to `() => void 0;` to get rid of that annoying error, or do what you did above AFAIK either is grand :)